packages/aws-cdk-lib/aws-eks/lib/service-account.ts

```diff
@@ -91,19 +163,61 @@ export class ServiceAccount extends Construct implements IPrincipal {
       throw RangeError('All namespace names must be valid RFC 1123 DNS labels.');
     }
 
-    /* Add conditions to the role to improve security. This prevents other pods in the same namespace to assume the role.
-    * See documentation: https://docs.aws.amazon.com/eks/latest/userguide/create-service-account-iam-policy-and-role.html
-    */
-    const conditions = new CfnJson(this, 'ConditionJson', {
-      value: {
-        [`${cluster.openIdConnectProvider.openIdConnectProviderIssuer}:aud`]: 'sts.amazonaws.com',
-        [`${cluster.openIdConnectProvider.openIdConnectProviderIssuer}:sub`]: `system:serviceaccount:${this.serviceAccountNamespace}:${this.serviceAccountName}`,
-      },
-    });
-    const principal = new OpenIdConnectPrincipal(cluster.openIdConnectProvider).withConditions({
-      StringEquals: conditions,
-    });
-    this.role = new Role(this, 'Role', { assumedBy: principal });
+    let principal: IPrincipal;
+    if (props.identityType !== IdentityType.POD_IDENTITY) {
+      /* Add conditions to the role to improve security. This prevents other pods in the same namespace to assume the role.
+      * See documentation: https://docs.aws.amazon.com/eks/latest/userguide/create-service-account-iam-policy-and-role.html
+      */
+      const conditions = new CfnJson(this, 'ConditionJson', {
+        value: {
+          [`${cluster.openIdConnectProvider.openIdConnectProviderIssuer}:aud`]: 'sts.amazonaws.com',
+          [`${cluster.openIdConnectProvider.openIdConnectProviderIssuer}:sub`]: `system:serviceaccount:${this.serviceAccountNamespace}:${this.serviceAccountName}`,
+        },
+      });
+      principal = new OpenIdConnectPrincipal(cluster.openIdConnectProvider).withConditions({
+        StringEquals: conditions,
+      });
+
+    } else {
+      /**
+       * Identity type is POD_IDENTITY.
+       * Create a service principal with "Service": "pods.eks.amazonaws.com"
+       * See https://docs.aws.amazon.com/eks/latest/userguide/pod-identities.html
+       */
+
+      // EKS Pod Identity does not support Fargate
+      if (cluster instanceof FargateCluster) {
+        throw Error('Pod Identity is not supported in Fargate. Use IRSA identity type instead.');
+      }
+      principal = new ServicePrincipal('pods.eks.amazonaws.com');
+    }
+
+    const role = new Role(this, 'Role', { assumedBy: principal });
+
+    // pod identities requires 'sts:TagSession' in its principal actions
+    if (props.identityType === IdentityType.POD_IDENTITY) {
+      /**
+       * EKS Pod Identities requires both assumed role actions otherwise it would fail.
+       */
+      role.assumeRolePolicy!.addStatements(new PolicyStatement({
+        actions: ['sts:AssumeRole', 'sts:TagSession'],
+        principals: [new ServicePrincipal('pods.eks.amazonaws.com')],
+      }));
+
+      // ensure the pod identity agent
+      cluster.eksPodIdentityAgent;
```

Comment:
what does line 208 do?

Reply:
it seems there is no permission assigned to the role?

Reply:
@nnmin-aws 

>Line 208

https://github.com/aws/aws-cdk/blob/8ef51b79e5573fc9cfc1abc4f35912a155bfe633/packages/aws-cdk-lib/aws-eks/lib/service-account.ts#L207-L208

This will execute this getter function:

https://github.com/aws/aws-cdk/blob/8ef51b79e5573fc9cfc1abc4f35912a155bfe633/packages/aws-cdk-lib/aws-eks/lib/cluster.ts#L2012-L2021

1. check if this cluster already has a eks pod identity agent created with it
2. if not, create one for it and assigned to `cluster. _eksPodIdentityAgent` attribute
3. this ensure a cluster would only run one single eks pod identity agent AddOn

Reply:
>it seems there is no permission assigned to the role?

When `ServiceAccount` is created, a role would be created with it. 

https://github.com/aws/aws-cdk/blob/8ef51b79e5573fc9cfc1abc4f35912a155bfe633/packages/aws-cdk-lib/aws-eks/lib/service-account.ts#L195

And if this ServiceAccount is for POD_IDENTITY, the assume role policy for it would be generated

https://github.com/aws/aws-cdk/blob/8ef51b79e5573fc9cfc1abc4f35912a155bfe633/packages/aws-cdk-lib/aws-eks/lib/service-account.ts#L202-L205

This role would be exposed to the ServiceAccount construct as a attribute:

https://github.com/aws/aws-cdk/blob/8ef51b79e5573fc9cfc1abc4f35912a155bfe633/packages/aws-cdk-lib/aws-eks/lib/service-account.ts#L220

User would be allowed to attach any policies or grant any policies to it.

For example:

```ts
// create a s3.Bucket;
const bucket = new s3.Bucket();
// create a service account
const sa = new ServiceAccount();
// allow the role of this service account to read/write the bucket
bucket.grantReadWrite(sa);
```